internal/tracegen/worker.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracegen
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"go.uber.org/zap"
+)
+
+type worker struct {
+	id       int             // worker id
+	traces   int             // how many traces the worker has to generate (only when duration==0)
+	marshall bool            // whether the worker needs to marshall trace context via HTTP headers
```

Comment:
not sure why this is necessary for a tracegen tool

Reply:
BTW marshall is used in various places in Jaeger code and I believe it is a typo. Marshal is the correct spelling AFAIK.